### PR TITLE
Removed Steam client dependency; Added replay name templates; Added option to save replays to specific path;

### PR DESCRIPTION
**Pull request by tynidev**

### CODE REVIEW

**Comment by tynidev** [comment]
Please squash these commits when you merge if possible @Bakkes :)

File: autoreplayuploader.set [hunk 1]
```diff
@@ -9,4 +9,8 @@ Auto replay uploader
 7|
```
Comment (tynidev) [inline autoreplayuploader.set]: Need to make one more change to this before this PR is complete.

**Comment by Bakkes** [comment]
Awesome! That's a lot of commits lol. Do you think it's ready? Has it been tested by multiple people without issues?

**Comment by Bakkes** [comment]
Can just ignore my code review comments, think its ready to merge if you think it's ready.

**Comment by tynidev** [comment]
I've had multiple people test it and I've resolved the issues so far.  But that really means like 3 other people besides me so that's not a very large user base.  I've been using it for a week plus and no issues so if there are issues it's likely I missed something in the user input section.  But I did try to unit test that so...

What do you think should I try and get more test users?

**Comment by Bakkes** [comment]
A few more users might be helpful. I'll try using it (if I ever play RL that is) and send it to some mods. If no issues we could probably include it as the default auto uploader plugin.

**Comment by tynidev** [comment]
Wait a second Bakkes doesn't play RL anymore?!?! Yeah I'm fine with either to be honest.  Right now I'm getting the value I wanted out of it myself which is really why I added it.  I'd rather not screw up other people's experience so if it causes issues at all I'd rather we just don't include it.

**Comment by Bakkes** [comment]
Lol. I log about an hour a week so I do play some RL. Anyway, there's an update tomorrow, so it probably won't make that. I do like it if this made the main release some day though since the replay template stuff has been requested multiple times. Have you seen the #bakkesplugins-beta stuff in the discord? Maybe once that's released we can put your version on there and easily get users to test it that way. Then if it's tested enough we can swap out the original one.

There's still one thing I would like to change though I just thought of. When the ballchasing auth key gets changed it does a lookup to the ballchasing API. This is good, but I don't want users that don't consent to ping a third party website. Right now, even if the user is not using the replay uploader plugin it will still do a request to ballchasing every time the game is launched (since from the config the command `cl_ballchasing_authkey ""` will be called, thus triggering a cvarchange). Maybe skip API key lookup if the new cvar value is under X amount of characters?

**Comment by tynidev** [comment]
Yeah I agree I think it is an improvement and I would like to get it in if
we can.  I'll keep testing it and if you could find some mods to help test
and give us confidence that would be awesome.

Agreed and you're right on startup it will try and validate even if it's
empty.  We could just do if it's an empty string don't test.  Then if the
users puts something there that isn't empty the assumption can be that they
want to test?

Also I have been wanting to add drop downs for common templates AND common
file save locations.  Then have a custom in each of those drop downs that
lets people define their own.  This would help people know what is
available and let people select things quickly in addition to being able to
customize it.

Tyler

**Pull request merged**